Program.cs

```diff
@@ -1,20 +1,103 @@
 ﻿// ---------------------------------------------------------------------------------------
 // Training ~ A training program for new joinees at Metamation, Batch- July 2023.
 // Copyright (c) Metamation India.                                                
-// ---------------------------------------------------------------------------
+// ---------------------------------------------------------------------------------------
 // Program.cs                                                                     
-// 
-// 
-// 
+// Program to create a custom MyList<T> class.
+// Class is created with arrays as the underlying data structure.
+// Must start with an initial capacity of 4 and double its capacity when needed.
+// Throw exceptions when necessary. 
 // ---------------------------------------------------------------------------------------
-using static System.Console;
+
 namespace Training {
-   #region class Program ------------------------------------------------------------------------
+   #region class Program --------------------------------------------------------------------------
    internal class Program {
-      #region Method ----------------------------------------------
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
+      #region Method ------------------------------------------------
+      static void Main () { }
+      #endregion
+   }
+
+   /// <summary>Custom MyList<T> class</summary>
+   public class MyList<T> {
+      #region Properity ---------------------------------------------
+      /// <summary>Gets the count of the MyList</summary>
+      public int Count => mCount;
+
+      /// <summary>Gets the size of the MyList</summary>
+      public int Capacity => mList.Length;
+
+      /// <summary>Gets the element or sets the value in the given index of MyList</summary>
+      /// <param name="index">Any index of MyList</param>
+      public T this[int index] {
+         get => !(index >= 0 && index < Count) ? throw new IndexOutOfRangeException () : mList[index];
+         set {
+            IdxOutOfRangeCheck (index);
+            mList[index] = value;
+         }
+      }
+      #endregion
+
+      #region Method ------------------------------------------------
+      /// <summary>This method adds a new element to the MyList</summary>
+      /// <param name="a">New element to be added</param>
+      public void Add (T a) {
+         if (Count == Capacity) Array.Resize (ref mList, Capacity * 2);
+         mList[mCount++] = a;
+      }
+
+      /// <summary></summary>
+      /// <param name="a">Any element in MyList</param>
+      /// <returns>Returns true if removed successfully</returns>
+      /// <exception cref="Exception">Throws exception when non-existing element is asked to remove</exception>
+      public bool Remove (T a) {
+         if (!mList.Contains (a)) return false;
+         int index = Array.IndexOf (mList, a);

```

Comment:
You are enumerating through the list twice here; first if the list contains the element to be removed and then again to get the index if it is present. Checking for an element through a list is of the order O(n) and same is the case with getting the index. To simplify it, you can just check for the index and if the element is not present in the list, the returned value will be -1. Then you can return accordingly.